src/NN_SDE_solve.jl

```diff
@@ -1,31 +1,166 @@
+                 learnt along with parameters of the neural network.
+
+* `dataset`: A dataset used to train the SDEPINN using observed process samples, at respective timepoints.
+             The `L2` loss, `moment_loss` is created using this. It is a Vector of `x`, `t` where `x` is a nested vector of multiple observations of the adapted process being learnt,
+             here each inner vector corresponds to one strong solution set of timeseries observations. `t` is a vector of timepoints at which we have the multiple `x` observations.
+
+* `data_sub_batch`:  The number of sets of random coefficients z_i to be taken for each timepoint while matching moments in `moment_loss`.
+                     The moment matching is done against strong process observations, therefore to match moments we must have corresponding SDEPINN outputs and inputs.
+                     Since realistically we cant get the eigenfunction, random coefficient decompositions for each observation we try to approximate the final moments by taking mean/sum
+                     over multiple sets of z_i concatenated with timepoints as SDEPINN inputs to get outputs that we can finally match with the dataset mean and variance.
+                     Is internally `max(data_sub_batch, length(dataset[1]))` by default.
+
+* `numensemble`: The solver returns an ensemble results/weak solution over user's provided saveat discretization. `numensemble` controls the Number
+                 of solution predictions to take an ensemble over for each timepoints. Is `10` by default.
+
+* `additional_loss`: A function additional_loss(phi, θ) where phi are the neural network
+                     trial solutions, θ are the weights of the neural network(s).
+
+* `kwargs`: Extra keyword arguments are splatted to the Optimization.jl `solve` call.
+
+## Examples
+
+```julia
+u0 = [1.0, 1.0]
+ts = [t for t in 1:100]
+(u_, t_) = (analytical_func(ts), ts)
+function additional_loss(phi, θ)
+    return sum(sum(abs2, [phi(t, θ) for t in t_] .- u_)) / length(u_)
+end
+alg = NNSDE(chain, opt, additional_loss = additional_loss)
+```
+
+```julia
+u₀ = 0.5
+f(u, p, t) = 1.2 * u
+g(u, p, t) = 1.1 * u
+tspan = (0.0, 1.0)
+prob = SDEProblem(f, g, u₀, tspan)
+n_z = 3
+dim = 1 + n_z
+luxchain = Chain(Dense(dim, 16, σ), Dense(16, 16, σ), Dense(16, 1))
+opt = BFGS()
+sol = solve(prob, NNSDE(luxchain, opt), verbose = true, dt = 1 / 50.0f0, abstol = 1e-10, maxiters = 200)
+```
+
+## Solution Notes
+
+Note that the returned weak solution is evaluated at fixed time points according to standard output
+handlers such as `saveat` and `dt`. However, the neural network is a fully continuous
+solution so `sol(t)` is an accurate interpolation (up to the neural network training
+result). In addition, the `OptimizationSolution` is returned as `sol.k` for further
+analysis.
+
+## References
+
+Stochastic Physics-Informed Neural Ordinary Differential Equations : https://arxiv.org/abs/2109.01621
```

Comment:
I'm not sure this reference makes sense since you're not doing a moment closure type of approach.